internal/resource/configmap.go

```diff
@@ -208,7 +208,20 @@ func (builder *ServerConfigMapBuilder) Update(object client.Object) error {
 		configMap.Data = make(map[string]string)
 	}
 
-	configMap.Data["rabbitmq.conf"] = rmqConfBuffer.String()
+	configMap.Data["operatorDefaults.conf"] = rmqConfBuffer.String()
+
+	rmqConfBuffer.Reset()
+
+	rmqProperties := builder.Instance.Spec.Rabbitmq
+	if err := cfgUserConfiguration.Append([]byte(rmqProperties.AdditionalConfig)); err != nil {
+		return fmt.Errorf("failed to append spec.rabbitmq.additionalConfig: %w", err)
+	}
+
+	if _, err := cfgUserConfiguration.WriteTo(&rmqConfBuffer); err != nil {
+		return err
+	}
+
+	configMap.Data["additionalConfig.conf"] = rmqConfBuffer.String()
 
 	updateProperty(configMap.Data, "advanced.config", rmqProperties.AdvancedConfig)
```

Comment:
The `additionalConfig` can also be empty right? If users do not configure `spec.rabbitmq.additionalConfig` and do not use TLS or additional plugins. Should we use `updateProperty()` to delete key `additionalConfig.conf` if it's empty?

Reply:
Good catch, the additional config can potentially be empty. Adding conditional logic here will have implications in the StatefulSet builder. Given that RabbitMQ can handle having an empty file and in the spirit of #KeepItSimple, I'd say we leave it as it is.

Reply:
Sounds good!